packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol

```diff
@@ -465,20 +463,23 @@ contract OptimisticAsserter is OptimisticAsserterInterface, Lockable, Ownable, M
     }
 
     function _callbackOnAssertionResolve(bytes32 assertionId, bool assertedTruthfully) internal {
-        if (assertions[assertionId].callbackRecipient != address(0))
-            OptimisticAsserterCallbackRecipientInterface(assertions[assertionId].callbackRecipient)
-                .assertionResolvedCallback(assertionId, assertedTruthfully);
-        if (assertions[assertionId].escalationManagerSettings.escalationManager != address(0))
-            EscalationManagerInterface(assertions[assertionId].escalationManagerSettings.escalationManager)
-                .assertionResolvedCallback(assertionId, assertedTruthfully);
+        address callbackRecipient = assertions[assertionId].callbackRecipient;
+        address escalationManager = _getEscalationManager(assertionId);
+        if (callbackRecipient != address(0))
+            OptimisticAsserterCallbackRecipientInterface(callbackRecipient).assertionResolvedCallback(
```

Comment:
equally, we could set the value within the assertion asserter here: https://github.com/UMAprotocol/protocol/blob/1d416c950a85fbb83df128d18b6503f0213c2a4f/packages/core/contracts/optimistic-asserter/interfaces/OptimisticAsserterInterface.sol#L26 to be of type `OptimisticAsserterCallbackRecipientInterface` we could also do the same by setting the address of the EM here https://github.com/UMAprotocol/protocol/blob/1d416c950a85fbb83df128d18b6503f0213c2a4f/packages/core/contracts/optimistic-asserter/interfaces/OptimisticAsserterInterface.sol#L12 to be of type EM.

Thoughts?

Reply:
I would rather not change OA interface. If we try avoiding casting it would also imply using callback recipient and EM interfaces in function parameters, thus requiring requesting contracts to import them, even if requesting contracts are not directly interacting with EM